ISTIO-SECURITY-2023-005 & CNI Docs Updates (#14302)
istio/istio.io: content/en/docs/setup/additional-setup/cni/index.md

To mitigate the race between an application pod and the Istio CNI DaemonSet,
an `istio-validation` init container is added as part of the sidecar injection,
which detects if traffic redirection is set up correctly, and blocks the pod starting up if not.
The CNI DaemonSet will detect and handle any pod stuck in such state; how the pod is handled is dependent on configuration described below.
This mitigation is enabled by default and can be turned off by setting `values.cni.repair.enabled` to false.

This repair capability can be further configured with different RBAC permissions to help mitigate the theoretical attack vector detailed in [`ISTIO-SECURITY-2023-005`](/news/security/istio-security-2023-005/).  By setting the below fields to true/false as required, you can select the Kubernetes RBAC permissions granted to the Istio CNI.

|Configuration                    | Roles       | Behavior on Error                                                                                                                           | Notes

> |---------------------------------|-------------|-----------------------------------------------------------------------------------------------------------------------------------------------|-------
> |`values.cni.repair.deletePods`   | DELETE pods | Pods are deleted, when rescheduled they will have the correct configuration       .                                                           | Default in 1.20 and older

Reviewer: ```suggestion
|---------------------------------|-------------|-----------------------------------------------------------------------------------------------------------------------------------------------|-------
|`values.cni.repair.deletePods`   | DELETE pods | Pods are deleted, when rescheduled they will have the correct configuration.                                                                  | Default in 1.20 and older
```

> The CNI DaemonSet will detect and evict any pod stuck in such state. When the new pod starts up, it should have traffic redirection set up properly.

Reviewer 2: ```suggestion
The CNI DaemonSet will detect and handle any pod stuck in such state; how the pod is handled is dependent on configuration described below.
```

(then see my next comment)

This mitigation is enabled by default and can be turned off by setting `values.cni.repair.enabled` to false.

> This repair capability can be further configured with different RBAC permissions to help mitigate the theoretical attack vector detailed in [`ISTIO-SECURITY-2023-005`](/news/security/istio-security-2023-005/) .  By setting the below fields to true/false as required, you can select the Kubernetes RBAC permissions granted to the Istio CNI.

Reviewer 3: ```suggestion
This repair capability can be further configured with different RBAC permissions to help mitigate the theoretical attack vector detailed in [`ISTIO-SECURITY-2023-005`](/news/security/istio-security-2023-005/).  By setting the below fields to true/false as required, you can select the Kubernetes RBAC permissions granted to the Istio CNI.
```

> |Configuration                    | Roles       | Notes

Reviewer 2: Maybe we should have another column on how it works for each mode:

delete: pods are deleted, and when rescheduled they will have the correct configuration
label: pods are only labeled. This mode requires a user to detect when pods with these labels are set and take manual action (typically, deleting the pod)
repair: pods are dynamically reconfigured to have appropriate configuration. When the container restarts, the pod will continue normal execution.

Reviewer 3: @pmerrison Thoughts on John's comment.
